lib/beaker/dsl/helpers/host_helpers.rb

```diff
@@ -224,6 +224,45 @@ def rsync_to host, from_path, to_path, opts = {}
           end
         end
 
+        # Copy a remote file to the local system and save it under a directory
+        # meant for storing SUT files to be viewed in the event of test failures.
+        #
+        # Files are stored locally with the following structure:
+        #   ./<archive_root>/<hostname><from_path>
+        #
+        # This can be used during the post-suite phase to persist relevant log
+        # files from the SUTs so they can available with the test results
+        # (without having to preserve the SUT host and SSH in after the fact).
+        #
+        # Example
+        #
+        #   Archive the Puppet Server log file from the master ('abc123'),
+        #   and archive the Puppet Agent log file from the agent ('xyz098'):
+        #
+        #     archive_file_from(master, '/var/log/puppetlabs/puppetserver.log')
+        #     archive_file_from(agent, '/var/log/puppetlabs/puppetagent.log')
+        #
+        #   Results in the following files on the test runner:
+        #
+        #     archive/sut-files/abc123/var/log/puppetlabs/puppetserver.log
+        #     archive/sut-files/xyz098/var/log/puppetlabs/puppetagent.log
+        #
+        # @param [Host] host A host object (or some object that can be passed to
+        #                    #scp_from)
+        # @param [String] from_path A remote absolute path on the host to copy.
+        # @!macro common_opts
+        # @options [String] archive_root The local directory to store the copied
+        #                                file under. Defaults to
+        #                                'archive/sut-files'.
+        #
+        # @return [Result] Returns the result of the #scp_from operation.
+        def archive_file_from(host, from_path, opts = {}, archive_root = 'archive/sut-files')
```

Comment:
We don't _need_ to make the archive root configurable, but I thought it might be useful when running tests locally if you wanted to route them to a different location.

But, this means we have two arguments with default values: `opts` and `archive_root`.
Assuming users aren't on newer Ruby versions that support keyword arguments...I decided to put go against the grain and make `archive_root` the last argument rather than `opts`.
This is to prevent users having to repeat the sensitive string `archive/sut-files` in the case when they want to provide some `opts` (which just go to the `scp_from` method).  This default value is important because that's what all our CI job-templates will be hardcoded to look for.

But maybe we don't need to make the archive root configurable?  Is it worth it?
But maybe we don't need to have an `opts` parameter either? I added it because I could imagine users wanting to be able to control the underlying `scp_from` for things like logging.